source/dpp/runtime/options.d

```diff
@@ -68,17 +69,46 @@ struct Options {
             args[1..$].filter!(a => a.extension != ".dpp").array ~
             dFileNames;
 
+        // use msvcrtd.lib if no other libc is specified
+        bool addNODEFAULTLIB = false;
+        version(Windows) {
+            if(mscrtlib) {
+                if(dlangCompiler == "dmd") {
+                    dlangCompilerArgs ~= mscrtlib ~ ".lib";
+                    addNODEFAULTLIB = true;
+                } else if (dlangCompiler == "ldc2") {
+                    dlangCompilerArgs ~= "--mscrtlib=" ~ mscrtlib;
+                }
+            }
+        } else {
+            assert(mscrtlib == "", "Using --mscrtlib flag on a non-windows platform is not allowed.");
+        }
+
         // if no -of option is given, default to the name of the .dpp file
-        if(!dlangCompilerArgs.canFind!(a => a.startsWith("-of")) && !dlangCompilerArgs.canFind("-c"))
-            dlangCompilerArgs ~= "-of" ~
+        if(!dlangCompilerArgs.canFind!(a => a.startsWith("-of")) && !dlangCompilerArgs.canFind("-c")) {
+            dlangCompilerArgs ~= ((dlangCompiler == "ldc2") ? "--of=" : "-of=") ~
                 args.
                 filter!(a => a.extension == ".dpp" || a.extension == ".d")
                 .front
                 .stripExtension
                 ~ exeExtension;
+        }
 
-        version(Windows)
+        version(Windows) {
+            // append the arch flag if not provided manually
+            if(!dlangCompilerArgs.canFind!(a => a == "-m64" || a == "-m32" || a == "-m32mscoff")) {
+                version(X86_64) {
+                    dlangCompilerArgs ~= "-m64";
+                }
+                version(x86) {
+                    dlangCompilerArgs ~= (dlangCompiler == "dmd") ? "-m32mscoff" : "-m32";
+                }
+            }
```

Comment:
Should we run this code on all operating systems? Architecture is independent of Windows.